subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdaterTest.groovy

```diff
@@ -52,20 +53,15 @@ abstract class AbstractFileWatcherUpdaterTest extends Specification {
     Predicate<String> watchFilter = { path -> !ignoredForWatching.contains(path) }
     def directorySnapshotter = new DirectorySnapshotter(TestFiles.fileHasher(), new StringInterner(), [])
     FileWatcherUpdater updater
-    def virtualFileSystem = new AbstractVirtualFileSystem() {
-        private SnapshotHierarchy root = DefaultSnapshotHierarchy.empty(CaseSensitivity.CASE_SENSITIVE)
+    def virtualFileSystem = new TestVirtualFileSystem(DefaultSnapshotHierarchy.empty(CaseSensitivity.CASE_SENSITIVE)) {
         @Override
-        SnapshotHierarchy getRoot() {
-            return root
-        }
-
-        @Override
-        protected void update(AbstractVirtualFileSystem.UpdateFunction updateFunction) {
+        protected SnapshotHierarchy updateNotifyingListeners(Function<SnapshotHierarchy.NodeDiffListener, SnapshotHierarchy> updateFunction) {
             def diffListener = new SnapshotCollectingDiffListener()
-            root = updateFunction.update(root, diffListener)
+            def newRoot = updateFunction.apply(diffListener)
```

Comment:
It's a bit confusing that we have both a `Function` and an `UpdateFunction` called `updateFunction` close to each other with the same functionality.

Reply:
Do you have a suggestion how to name those things?

Reply:
I removed `UpdateFunction` and the `update` method. Using `updateNotifyingListeners` isn't too ugly.